symengine/polys/uexprpoly.cpp

```diff
@@ -10,7 +10,16 @@ namespace SymEngine
 UExprPoly::UExprPoly(const RCP<const Symbol> &var, UExprDict &&dict)
     : UPolyBase(var, std::move(dict))
 {
-    SYMENGINE_ASSERT(is_canonical(poly_))
+    auto iter = dict.dict_.begin();
+    while (iter != dict.dict_.end()) {
+        if (iter->second == Expression(0)) {
+            auto toErase = iter;
+            iter++;
+            dict.dict_.erase(toErase);
+        } else {
+            iter++;
+        }
+    }
```

Comment:
You should understand the purpose of checking canonicalisation in debug builds only.
Checking that the polynomial is canonical is expensive. You don't need to check that it is not canonical because in all the operations, you are making a canonical dictionary. `SYMENGINE_ASSERT` is there to check this in Debug builds and it doesn't do anything in Release builds